src/formats/apple-notes/convert-note.ts

```diff
@@ -0,0 +1,398 @@
+				
+				row = await this.importer.database.get`
+					SELECT z_pk FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${destIdentifier.toUpperCase()}`;
+					
+				if (!(row.Z_PK in this.importer.resolvedFiles)) {
+					//if we don't have the note yet, we fetch it early to get its path
+					this.importer.resolveNote(row.Z_PK);
+				}
+				
+				return this.app.fileManager.generateMarkdownLink(
+					this.importer.resolvedFiles[row.Z_PK] as TFile, 
+					this.importer.rootFolder.path
+				);
+				
+			case ANAttachment.Table:
+				row = await this.importer.database.get`
+					SELECT hex(zmergeabledata1) as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+
+				converter = this.importer.decodeData(row.zhexdata, TableConverter);
+				return await converter.format();
+			
+			case ANAttachment.UrlCard:
+				row = await this.importer.database.get`
+					SELECT ztitle, zurlstring as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+					
+				return `[**${row.ZTITLE}**](${row.ZURLSTRING})`;
+			
+			case ANAttachment.Scan:
+				row = await this.importer.database.get`
+					SELECT hex(zmergeabledata1) as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				converter = this.importer.decodeData(row.zhexdata, ScanConverter);
+				return await converter.format();
+			
+			case ANAttachment.ModifiedScan:
+			case ANAttachment.Drawing:
+				row = await this.importer.database.get`
+					SELECT z_pk FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				id = row?.Z_PK;
+				break;
+				
+			//actual file on disk (eg image, audio, video, pdf, vcard)
+			//hundreds of different utis so not in the enum
+			default:
+				row = await this.importer.database.get`
+					SELECT zmedia FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				id = row?.ZMEDIA;
+				break;
+		}
+		
+		if (!id) {
+			//doesn't have an associated file, so unknown
+			return ` **(unknown attachment: ${attr.attachmentInfo.typeUti})** `;
+		}
+		
+		await this.importer.resolveAttachment(id, attr.attachmentInfo.typeUti);
+		const link = this.app.fileManager.generateMarkdownLink(
+			this.importer.resolvedFiles[id] as TFile, '/'
+		);
+		
+		return `\n${link}\n`;	
+	}
+	
+	convertColor(color: ANColor): string {
+		let hexcode = '#';
+		
+		for (const channel of Object.values(color)) {
+			hexcode += Math.floor(channel * 255).toString(16);
+		}
+		
+		return hexcode;
+	}
+	
+	convertAlign(alignment: ANAlignment): string {
+		switch (alignment) {
+			default: return 'left';
+			case ANAlignment.Centre: return 'center';
+			case ANAlignment.Right: return 'right';
+			case ANAlignment.Justify: return 'justify';
+		}
+	}
+}
+
+function isBlockAttachment(attr: ANAttributeRun) {
```

Comment:
`attr` is never null/undefined here, right? Which means we probably don't need the `?`s down below

Reply:
That's incorrect, attr is null at the end. But it makes sense to check for that explicitly, so I've changed it so it does that.

Reply:
If it can be null then the typescript definition should have been `attr: ANAttributeRun | null`.